prometheus_juju_backup_all_exporter/utils.py

```diff
@@ -73,37 +73,37 @@ def result_code(self) -> int:
         return self._result_code
 
 
-class BackupState:
-    """A class representing backup state file."""
+class BackupEvent:
+    """A class representing backup event file."""
 
     def __init__(self, config: Config) -> None:
         """Initialize and set instance properties."""
         self._failed = DEFAULT_FAILED
         self._purged = DEFAULT_PURGED
         self._completed = DEFAULT_COMPLETED
-        state_file = Path(config.backup_path, "backup_state.json")
+        event_file = Path(config.backup_path, "backup_state.json")
```

Comment:
Since we are doing refactor, can we change following lines of code to:

```python
event_file = Path(config.backup_path, "backup_state.json")
if not event_file.exists():
    logger.warning(
        "Backup event file: %s does not exist, using default values.", str(event_file),
    )
    return

try:
    with open(event_file, "r", encoding="utf-8") as event:
        backup_event = json.load(event)
        self._failed = backup_event["failed"]
        self._purged = backup_event["purged"]
        self._completed = backup_event["completed"]
except (KeyError, PermissionError, json.decoder.JSONDecodeError) as err:
    logger.error(
        "Invalid backup event file: %s. %s. Using default values.",
        str(event_file),
        str(err),
    )
finally:
    logger.info("Removing event file: %s.", str(event_file))
    event_file.unlink()
```

Reply:
Thanks, I've updated this part.